docs/applications.md

```diff
@@ -175,6 +175,29 @@ An example `data` dictionary to specify on `POST /_matrix/client/r0/pushers/set`
 [APNs documentation]: https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CreatingtheNotificationPayload.html
 
 
+#### VoIP (voice calls and video calls) notifications on iOS
+
+iOS is capable of displaying an on-screen call notification (with answer/deny buttons)
+and playing a ringtone.
+
+However, this requires sending a special kind of push notification.
+Sygnal is not able to send this type of push notification because a VoIP call may
+begin as a result of an encrypted message, so Sygnal has no way to know that a given
+message should initiate a VoIP call.
+
+The current workaround is for your iOS application to register a Notification
+Service Extension and to use that Notification Service Extension to trigger a
+VoIP push notification with Sygnal when your application needs to display a call
+notification.
```

Comment:
I'm not even sure this is a workaround; IIRC this is intended on Apple's part.

I also vaguely remember the following from a previous role: If you send a voip call notification and don't present a CallKit UI within ~30seconds, then future voip call notifications will get delayed or throttled. I don't think this was a concretely documented thing other than "if you abuse this, we'll punish you".

I think links to Apple's documentation would be appropriate, e.g. https://developer.apple.com/documentation/pushkit/responding_to_voip_notifications_from_pushkit

Reply:
> If you send a voip call notification and don't present a CallKit UI within ~30seconds

I might have confused this for the section called Wake Your App with a Background Push [here](
https://developer.apple.com/documentation/backgroundtasks/choosing_background_strategies_for_your_app#overview).

Reply:
> IIRC this is intended on Apple's part.

I would have thought the intended way would be to directly send the VoIP notifications to the app rather than have the app re-request one on demand

Reply:
Oh sorry, I didn't read properly. The use of the NSX to intercept and post-process/handle notification is definitely intentional; I didn't realise you were proposing that there'd be another request and another round-trip to the server.